subprojects/core-api/src/main/java/org/gradle/api/file/ProjectLayout.java

```diff
@@ -63,6 +64,26 @@
      */
     FileCollection files(Object... paths);
 
+    /**
+     * <p>Creates a read-only {@code FileTree} which contains the contents of the given ZIP file, as defined by {@link Project#zipTree(Object)}.</p>
+     *
+     * @param zipPath The ZIP file. Evaluated as per {@link Project#file(Object)}.
+     * @return the file tree. Never returns null.
+     * @since 6.6
+     */
+    @Incubating
+    FileTree zipTree(Object zipPath);
```

Comment:
This doesn't really have anything to do with the project layout (which is concerned with querying certain key locations of a project). `ProjectLayout` also isn't available outside of project scoped build logic.

Some better places might be:
- On `RegularFile` (without the parameter)
- On `Directory` to live alongside `files()`.
- On `FileSystemOperations`
- On a new service interface that concerns itself with Zip and Tar operations.

Reply:
Moved to their own service in https://github.com/gradle/gradle/pull/13305
Convenience on `RegularFile` or `Directory` could be added later.